src/test/java/org/itsallcode/openfasttrace/importer/TestSpecificationListBuilder.java

```diff
@@ -172,4 +176,45 @@ public void testDuplicateIdNotIgnored()
         builder.endSpecificationItem();
         assertThat(builder.getItemCount(), equalTo(2));
     }
+
+    // [utest->dsn~filtering-by-tags-during-import~1]
+    @Test
+    public void testFilterSpecificationItemsByTags()
+    {
+        final SpecificationListBuilder builder = createListBuilderFilteringByTags("client",
+                "server");
+        builder.beginSpecificationItem();
+        final SpecificationItemId idA = SpecificationItemId.createId("dsn", "in-A", 1);
+        builder.setId(idA);
+        builder.addTag("client");
+        builder.addTag("database");
+        builder.endSpecificationItem();
+        builder.beginSpecificationItem();
+        final SpecificationItemId idB = SpecificationItemId.createId("dsn", "in-B", 1);
+        builder.setId(idB);
+        builder.addTag("server");
+        builder.addTag("database");
+        builder.endSpecificationItem();
+        builder.beginSpecificationItem();
+        final SpecificationItemId idC = SpecificationItemId.createId("dsn", "out-C", 1);
+        builder.setId(idC);
+        builder.addTag("exporter");
+        builder.addTag("database");
+        builder.endSpecificationItem();
+        builder.beginSpecificationItem();
+        final SpecificationItemId idD = SpecificationItemId.createId("dsn", "out-D", 1);
+        builder.setId(idD);
+        builder.endSpecificationItem();
+        final List<SpecificationItem> items = builder.build();
+        assertThat(items.stream().map(item -> item.getId().getName()).collect(Collectors.toList()),
+                containsInAnyOrder("in-A", "in-B"));
+    }
+
+    private SpecificationListBuilder createListBuilderFilteringByTags(final String... tags)
+    {
+        final FilterSettings filterSettings = new FilterSettings.Builder() //
+                .tags(new HashSet<>(Arrays.asList(tags))) //
```

Comment:
Overload `tags()` to accept `String... tags` and call `new HashSet<>(Arrays.asList(tags))` internally to make this shorter.